crates/npmrc/src/custom_deserializer.rs

```diff
@@ -16,6 +16,30 @@ pub fn default_public_hoist_pattern() -> Vec<String> {
     vec!["*eslint*".to_string(), "*prettier*".to_string()]
 }
 
+// Get the drive letter from a path on Windows. If it's not a Windows path, return None.
+fn get_drive_letter(current_dir: &Path) -> Option<String> {
+    for component in current_dir.components() {
```

Comment:
Drive letter will only ever exist as the very first component, so looping is sub-optimal.

```suggestion
    if let Some(component) = current_dir.components().next() {
```